Swift 4

chadcummings opened the pull request:
Swift 4 Upgrades! Let me know if I missed anything.

- Upgrades to project and code to allow for Swift 4, SDK iOS11.
- Target iOS 9.3.
- Add support for UPC-A.
- Add all new barcode types to the metadata array.

vadymmarkov commented on Sources/BarcodeScannerController.swift:
```diff
@@ -343,8 +343,6 @@ open class BarcodeScannerController: UIViewController {
     }
     center(subview: settingsButton, inSize: CGSize(width: 150, height: 50))
-
-    headerView.isHidden = !isBeingPresented
```
@chadcummings What is the reason behind this change?

chadcummings replied:
This fixes a bug when the screen orientation was changed, the Header would disappear and never reappear, even when the device was rotated back. I haven't noticed any negative side effects, but I'll look to you for guidance here.

vadymmarkov replied:
I think it was here for the case when you have `UINavigationController` and push barcode scanner controller, so you don't end up with two "navigation bars". But arguably it's something you can manually control in the code.

chadcummings replied:
I'll see if I can test this out by enhancing the example app to verify. I'll fix it if necessary.

vadymmarkov replied:
@chadcummings Keep us posted. It's the only thing we'd like to verify before merging this PR.

vadymmarkov commented on BarcodeScanner.xcodeproj/project.pbxproj:
```diff
@@ -228,7 +234,7 @@
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.2;
+				IPHONEOS_DEPLOYMENT_TARGET = 9.3;
```
Could it be `9.0` or there are some specific APIs you want to use? @chadcummings

chadcummings replied:
After reviewing which devices can run iOS 9.0, I discovered that all of the target devices can also run iOS 9.3. This led me to the conclusion that is was pointless to support iOS 9.0 when all devices that run iOS 9.0 will run iOS 9.3. Supporting iOS 9.0 has no benefit, as no additional devices are gained. That's my only reasoning, really.

Ultimately, this code is your's. I am happy to use whichever version you would like, but I would prefer trimming extra baggage as much as possible.

vadymmarkov replied:
I totally agree with you, but I think we still have projects with minimum target set to 9.0, which is probably something we might want to change. What do you think @onmyway133 @zenangst?

onmyway133 replied:
I don't have strong opinion about this. But as there's no reason to raise to 9.3, then 9.0 is fine I think

chadcummings replied:
Sounds good. I'll move it to 9.0.

zenangst replied:
@vadymmarkov yeah we should totally change that. But if we don't need `9.3` changes in this PR then it should still use `9.0` :)

Patchivic commented:
great @chadcummings 
I liked it :)

zenangst commented:
I like this PR!

chadcummings commented:
Sorry, everyone. Work got a little busy for a bit.

I've changed the target to 9.0 and that all works well.

I enhanced the example project to demonstrate push and present navigation to ensure we didn't have a problem with "double navigation bars" with a line of code I removed.
It's working well with push navigation but for some reason the flashButton has now disappeared when pushing and presenting the BarcodeScannerController.

I'll see if I can get that fixed quick so we can finish up this PR.

chadcummings commented:
Ooookkaaayyy, That's embarrassing. Apparently my iPad physically doesn't have a a light with the camera. Which explains why the flashButton doesn't show up. Problem solved.

chadcummings commented:
Final fixes done. Let me know if there is anything else to do. Thanks.

vadymmarkov commented:
@chadcummings Awesome! Thanks for your contribution.

vadymmarkov commented:
Version 3.0.0 is out.
